db/migrate/20150828153331_change_business_support_edition_areas_to_gss_codes.rb

```diff
@@ -0,0 +1,22 @@
+class ChangeBusinessSupportEditionAreasToGssCodes < Mongoid::Migration
+  def self.up
+    # Editing of an edition with an Archived artefact is not allowed
+    Edition.skip_callback(:save, :before, :check_for_archived_artefact)
+
+    BusinessSupportEdition.all.each do |edition|
+      edition.area_gss_codes = Area.areas_for_edition(edition).map { |area|
+        area.codes["gss"]
+      }.compact
+
+      edition.save!(validate: false) # Published editions can't be edited.
+    end
+  end
+
+  def self.down
+    # If this needs rolling back perminantly, the area_gss_codes field should
+    # probably just be removed from GOV.UK Content Models.
+    #
+    # .up is only adding data, not changing existing data, so I assume
+    # a temporary rollback doesn't really require any action.
```

Comment:
I think we should remove the `area_gss_codes` field in the `down` method so that we can keep the data clean, rather than relying on ignoring extra fields in code if we need to roll back. We should be able to migrate the database back to its original state by rolling back this migration, and when we've switched over to using the GSS codes we should remove the `areas` field in another migration.